vignettes/test-children.Rmd

```diff
@@ -0,0 +1,165 @@
+---
+title: "Testing child packages"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Testing child packages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette is written for authors of packages that depend on crunch, who want
+to use similar testing infrastructure as the main crunch package. The crunch
+package itself has a long, evolutionary history, so not all of this advice applies 
+directly to developing for rcrunch itself 
+([more details at bottom of this document](#rcrunch-specific)). 
+
+When testing code that relies on an API, there are two broad categories of testing:
+
+- **Integration testing** - The tests call to the server each time they are run. 
+  While this is nice in theory, it can cause problems because test code can
+  overload the servers because of how frequently they need to run which will not only
+  cause your tests to fail, but negatively affect all users of the API.
+- **Mock testing** - Rather than actually call the server, the tests use "mocks"
```

Comment:
There is another type of testing here that's built into testthat called "with_mock" that can be extremely useful and in many cases better than trying to create fixtures. I have made some examples internally for the innovations team but I think they belong here as well. 

![image](https://user-images.githubusercontent.com/421859/92281211-b968b500-eec0-11ea-83b0-8b10326ab908.png)

Reply:
Generally as well the guidance to separate I/O from logic is a good idea. Many on our team are loading a ds than going directly to a data.frame (do not pass go ,do not collect $200). These calls are often mixed in with other logic that could be tested independently if they were indepdent of crunch.